akka-stream/src/main/scala/akka/stream/impl/ActorPublisher.scala

```diff
@@ -64,8 +64,7 @@ private[akka] class ActorPublisher[T](val impl: ActorRef) extends Publisher[T] {
 
   def takePendingSubscribers(): immutable.Seq[Subscriber[_ >: T]] = {
     val pending = pendingSubscribers.getAndSet(Nil)
-    assert(pending ne null, "takePendingSubscribers must not be called after shutdown")
-    pending.reverse
+    if (pending eq null) Nil else pending.reverse
```

Comment:
Please ponder this one: I cannot currently explain why this patch changed the behavior, and I also cannot explain why this previously worked with the assertion.